### PR TITLE
[ENH] Allow SurfaceLabelsMasker to work with a list of Surface images

### LINKED ISSUE #4762: run fit and transform of SurfaceLabelsMasker on list of Surface image
After that I think I will do something similar for the label masker.

_Originally posted by @Remi-Gau in https://github.com/nilearn/nilearn/issues/4719#issuecomment-2483234819_

### PR DESCRIPTION
**Pull request by man-shu**
- Closes #4762

Changes proposed in this pull request:

- use concatenation of a list of surface images and then do the masking
- allow inverse_transform method to work with 2D extracted region signals

### CODE REVIEW

**Comment by Remi-Gau** [comment]
> I don't quite understand the pre-release test failure. Any pointers, @Remi-Gau ?

it is unrelated to this PR as they are also present on main:
https://github.com/nilearn/nilearn/actions/runs/11929581563/job/33248657361

testing with scikit-learn==1.6.0rc1 leads to these failures

I think that it'd be good to have those fixed for the release.

Another possibility could be to pin scikit-learn to <1.6.0 for the coming release.

scikit-learn 1.6.0 should come out soon: see milestone https://github.com/scikit-learn/scikit-learn/milestone/57

But I am worried of release something that is not compatible with the latest stable scikit-learn

**Comment by man-shu** [comment]
> testing with scikit-learn==1.6.0rc1 leads to these failures

Ok cool! I thought so, but could not find the exact version of scikit-learn installed in the environment. Where do you see that? Or is it just always installed directly from the Github repo?

File: nilearn/maskers/surface_labels_masker.py [hunk 1]
```diff
@@ -362,9 +374,14 @@ def inverse_transform(self, masked_img):
...
-                    ..., label_idx
-                ]
+                if masked_img.ndim == 1:
+                    data[part_name][..., labels_part == label] = masked_img[
+                        ..., label_idx
+                    ]
+                else:
+                    data[part_name][..., labels_part == label] = masked_img[
+                        label_idx
+                    ]
```
[the remark this answers is not in the document]
Comment (man-shu) [reply nilearn/maskers/surface_labels_masker.py:384]: I think you do. The list test you added would take care of the 2D input, and there were specific tests for transform and inverse_transform for 1D input already.

**Comment by Remi-Gau** [comment]
- blocked by #4771

**Review by bthirion — APPROVED** [review]
LGTM, thx !

File: nilearn/maskers/tests/test_surface_labels_masker.py [hunk 2]
```diff
@@ -104,8 +104,8 @@ def test_surface_label_masker_transform(surf_label_img, surf_img):
     assert signal.shape == (n_timepoints, n_labels)
 
 
-def test_surface_label_masker_transform_check_output(surf_mesh, surf_img, rng):
-    """Check actual content of the transform.
+def test_surface_label_masker_check_output(surf_mesh, rng):
+    """Check actual content of the transform and inverse_transform.
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply nilearn/maskers/tests/test_surface_labels_masker.py:108]: ah nice catch and thanks for fixing

**Review by Remi-Gau — APPROVED** [review]

**Pull request merged**